giraffe/src/components/AnnotationTooltip.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import {FunctionComponent} from 'react'
+import {createPortal} from 'react-dom'
+
+import {Config, AnnotationMark, TooltipPosition} from '../types'
+import {ANNOTATION_TOOLTIP_CONTAINER_NAME} from '../constants'
+import {useTooltipElement} from '../utils/useTooltipElement'
+
+interface Props {
+  config: Config
+  data: AnnotationMark
+  boundingReference: DOMRect
+  width: number
+  height: number
+}
+
+export const AnnotationTooltip: FunctionComponent<Props> = props => {
+  const {config, data, boundingReference, width} = props
+  const {
+    legendFont: font,
+    legendFontColor: fontColor,
+    legendBackgroundColor: backgroundColor,
+    legendBorder: border,
+  } = config
+  const {dimension, startValue} = data || {}
+
+  const position = {
+    x: dimension === 'x' ? startValue : width,
+    y: dimension === 'y' ? startValue : 0,
+  } as TooltipPosition
+  const annotationTooltipElement = useTooltipElement(
+    ANNOTATION_TOOLTIP_CONTAINER_NAME,
```

Comment:
what's the thinking behind having this be a classname? is it better served as a data-attr? what is its main intent?

Reply:
`useTooltipElement` is an existing function that adds a classname to the tooltip's outermost element. Previously, it was hardcoded as "giraffe-tooltip-container". That class name isn't used for any styling in Giraffe itself as far as i can tell. But this was a nice thing that Giraffe exposed for consuming apps to use--should they need it.

Following this thinking, the change here was just to allow the classname to be given by the caller, since `useTooltipElement` is now a shared function for 2 different types of tooltips. We still want to have a classname for consuming apps to easily style the tooltip container with CSS targeting, should they prefer to go that route.

Reply:
>But this was a nice thing that Giraffe exposed for consuming apps to use--should they need it.

i agree, this is a nice thing to have for the future. i think since there's no styling of it, it most certainly shouldn't be a classname, it should be a data-attribute, which is the point of having data attributes - to keep things semantic. not a blocker, but something that should probably be changed for overall ease of comprehension